src/components/shared/globalTypes.ts

```diff
@@ -153,31 +156,44 @@ export class Directory extends FileSystemObject {
   changeCurrentWorkingDirectory(
     currentWorkingDirectory: Directory,
     path: string
-  ): Directory | undefined {
+  ): Directory | string {
     currentWorkingDirectory.isCurrentDirectory = false;
     let newCwd: Directory | File | undefined;
 
-    if (!path.startsWith('/')) {
-      newCwd = currentWorkingDirectory.getFileSystemObjectFromPath(path);
-    } else {
+    // If the path is absolute, we start from the root directory and find the new cwd
+    if (path.startsWith('/')) {
       newCwd = this.getFileSystemObjectFromPath(path);
+    } else {
+      newCwd = currentWorkingDirectory.getFileSystemObjectFromPath(path);
     }
 
-    if (!newCwd || !newCwd.isDirectory) return undefined;
+    // If the new cwd is a file or does not exist, we can't cd into it
+    if (!newCwd) {
```

Comment:
Couldn't these two `if` statements be combined into one?

Reply:
Hmm, I think no because the return is different for these two cases. It's quite similar, but the first is that no such directory exists, but the second is that the newCwd is simply not a director (it's a file with the same path). They use to be treated as the same case, but since the error is slightly different, the if statements should be separate.